# HG changeset patch
# User Rik <rik@octave.org>
# Date 1427598180 25200
#      Sat Mar 28 20:03:00 2015 -0700
# Node ID 9a2d8b1f931e3330d25a494dbe4e3cce0147b67c
# Parent  7569e880d56c2b162c5e402ad1b586567a82764f
Issue error when trying to print a figure with no axes object (bug #44655).

* print.m: Check for an axes object as a child of the figure object to be
printed.

diff --git a/scripts/plot/util/print.m b/scripts/plot/util/print.m
--- a/scripts/plot/util/print.m
+++ b/scripts/plot/util/print.m
@@ -296,16 +296,20 @@ function print (varargin)
   opts.latex_standalone = @latex_standalone;
   opts.lpr_cmd = @lpr;
   opts.epstool_cmd = @epstool;
 
   if (isempty (opts.figure) || ! isfigure (opts.figure))
     error ("print: no figure to print");
   endif
 
+  if (isempty (findall (opts.figure, "-depth", "1", "type", "axes")))
+    error ("print: no axes object in figure to print");
+  endif
+
   orig_figure = get (0, "currentfigure");
   set (0, "currentfigure", opts.figure);
 
   if (opts.append_to_file)
     [~, ~, ext] = fileparts (opts.ghostscript.output);
     opts.ghostscript.prepend = [tempname() ext];
     copyfile (opts.ghostscript.output, opts.ghostscript.prepend);
   endif
